### PR TITLE
Refactor TextWidget

### PR DESCRIPTION
**Pull request by Tishasoumya-02**
#4460

### CODE REVIEW

File: packages/volto/src/components/manage/Widgets/TextWidget.jsx [hunk 1]
```diff
@@ -17,120 +10,94 @@ import FormFieldWrapper from '@plone/volto/components/manage/Widgets/FormFieldWr
...
+    iconAction,
+    minLength,
+    maxLength,
+    placeholder,
+    isDisabled,
+  } = props;
+
+  const ref = useRef(null);
+
+  useEffect((focus) => {
```
Comment (sneridagh) [inline packages/volto/src/components/manage/Widgets/TextWidget.jsx:31]: @Tishasoumya-02 AFAIK, `useEffect` setup function does not take any arguments.

**Review by sneridagh — CHANGES_REQUESTED** [review]

File: packages/volto/src/components/manage/Widgets/TextWidget.jsx (hunk 1 above)
Comment (Tishasoumya-02) [reply packages/volto/src/components/manage/Widgets/TextWidget.jsx]: yes, I'm not sure why I did this, thank you for pointing out.

Comment (Tishasoumya-02) [reply packages/volto/src/components/manage/Widgets/TextWidget.jsx]: Not sure, why the cypress tests are failing, will look into it

File: packages/volto/news/6020.feature [hunk 2]
```diff
@@ -0,0 +1 @@
+Refactor TextWidget -Tishasoumya-02
```
Comment (davisagli) [inline packages/volto/news/6020.feature:1]:
```suggestion
Refactor TextWidget. @Tishasoumya-02
```

**Review by davisagli — APPROVED** [review]
Thanks!

**Pull request merged**